src/read_load_time_mpi_constants.jl

```diff
@@ -277,11 +279,287 @@ global MPI_WIN_NULL_COPY_FN = unsafe_load(cglobal((:MPICONSTANTS_WIN_NULL_COPY_F
 
 global MPI_WIN_NULL_DELETE_FN = unsafe_load(cglobal((:MPICONSTANTS_WIN_NULL_DELETE_FN, libmpiconstants), MPI_Win_delete_attr_function))
 
-else
+else # if Sys.iswindows()
 
-include("read_load_time_mpi_constants_msmpi.jl")
+# MicrosoftMPI only supports the standard MPI 2.0, so not all
+# constants are available. (Julia generally supports MPI 3.1.)
 
-end
+global MPI_ANY_SOURCE = Cint(-2)
```

Comment:
Is this not type-instable now?

Reply:
Which part? `MPI_ANY_SOURCE` is always of type `Cint`.

Do you refer to the fact that this is a global variable and not a constant, so that the compiler doesn't know the type won't change?

Reply:
> Do you refer to the fact that this is a global variable and not a constant, so that the compiler doesn't know the type won't change?

Yeah.

Reply:
We discussed this at some point. The best solution would be to change the constants' types to `Ref{Cint}` to make them constants. The (slight) disadvantage is then that accessing these constants requires dereferencing them. However, if I understand correctly, these constants are not supposed to be used from the outside anyway.

We can also move all the constants into a submodule `MPI.Consts`. Should I implement this? It looks straightforward.

Reply:
Yeah dereferencing is quite a bit cheaper. And having the constant be actual constant would be great as well.

We could also consider turning the non-constant globals into functions.

Reply:
We can't make them truly constant; Julia doesn't have the concept of constants that are defined at load time. If a value is declared constant, then it needs to be known while precompiling. One cannot declare constants in `__init__`.

Reply:
Done.